### PR TITLE
#1699 add sensible defaults to NullValuePropertyMapping.SET_TO_DEFAULT

### PR DESCRIPTION
**Pull request by sjaakd**
@filiphr .. can't get the javadoc to work.. Perhaps you spot the issue?

Ok.. Its the maven build that is a bit unclear. I spelt `{@Code xyz}` iso `{@code xyz}`

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_1685/UserMapper.java [hunk 1]
```diff
@@ -38,7 +38,6 @@
 
     @InheritInverseConfiguration
     @BeanMapping( nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.SET_TO_DEFAULT )
-    @Mapping( target = "phone", source = "contactDataDTO.phone", defaultValue = "0" )
```
Comment (filiphr) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_1685/UserMapper.java]: Shouldn't we just remove the `defaultValue`? At least it was like that before.

Comment (sjaakd) [reply processor/src/test/java/org/mapstruct/ap/test/bugs/_1685/UserMapper.java]: it's inherited.. To reassure you: the fixture did not change :smile:

Comment (filiphr) [reply processor/src/test/java/org/mapstruct/ap/test/bugs/_1685/UserMapper.java]: Ah yes I missed thw inheritance 😀

**Comment by sjaakd** [comment]
@filiphr .. thanks for reviewing..

**Pull request merged**